src/utils.coffee

```diff
@@ -130,3 +130,12 @@ exports.clone = (obj) ->
     copy[key] = exports.clone(obj[key])
 
   return copy
+
+# cut leading and trailings blanks 
+ 
+exports.trimWhitespace = (value) ->
+    result = value.match(/^\s*(.*\S)\s*$/);
+    if (result != null && result.length == 2)
+        return result[1]
+    else
+        return value
```

Comment:
Thanks for free course Noah ! I'll try to simplify and recommit as
expected  (not yet as i work on day and program on night...).
In fact i was comparing ibm.watson dialog architecture with rivescript.
First uses an ugly xml parsed in a black box. Yours is much much more
comfortable despite some missing features vs first. Thanks to be open.
Le 21 déc. 2015 22:45, "Noah Petherbridge" <notifications@github.com> a
écrit :

> In src/utils.coffee
> <https://github.com/aichaos/rivescript-js/pull/53#discussion_r48195996>:
>
> > @@ -130,3 +130,12 @@ exports.clone = (obj) ->
> >      copy[key] = exports.clone(obj[key])
> >
> >    return copy
> > +
> > +# cut leading and trailings blanks
> > +
> > +exports.trimWhitespace = (value) ->
> > +    result = value.match(/^\s*(.*\S)\s*$/);
> > +    if (result != null && result.length == 2)
> > +        return result[1]
> > +    else
> > +        return value
>
> This function isn't needed; use the built-in String.prototype.trim()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aichaos/rivescript-js/pull/53/files#r48195996>.
>